demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1685 @@
+
+    assert( ( otaMessageType >= 0 ) && ( otaMessageType < OtaNumOfMessageType ) );
+
+    /* Register callback to subscription manager. */
+    subscriptionStatus = SubscriptionManager_RegisterCallback( pTopicFilter,
+                                                               topicFilterLength,
+                                                               otaMessageCallback[ otaMessageType ] );
+
+    if( subscriptionStatus != SUBSCRIPTION_MANAGER_SUCCESS )
+    {
+        LogWarn( ( "Failed to register a callback to subscription manager with error = %d.",
+                   subscriptionStatus ) );
+    }
+
+    return otaRet;
+}
+
+/*-----------------------------------------------------------*/
+
+static OtaMqttStatus_t mqttPublish( const char * const pacTopic,
+                                    uint16_t topicLen,
+                                    const char * pMsg,
+                                    uint32_t msgSize,
+                                    uint8_t qos )
+{
+    OtaMqttStatus_t otaRet = OtaMqttSuccess;
+
+    MQTTStatus_t mqttStatus = MQTTBadParameter;
+    MQTTPublishInfo_t publishInfo;
+    MQTTContext_t * pMqttContext = &mqttContext;
+
+    /* Set the required publish parameters. */
+    publishInfo.pTopicName = pacTopic;
+    publishInfo.topicNameLength = topicLen;
+    publishInfo.qos = qos;
+    publishInfo.pPayload = pMsg;
+    publishInfo.payloadLength = msgSize;
+
+    if( pthread_mutex_lock( &mqttMutex ) == 0 )
+    {
+        mqttStatus = MQTT_Publish( pMqttContext,
+                                   &publishInfo,
+                                   MQTT_GetPacketId( pMqttContext ) );
+
+        pthread_mutex_unlock( &mqttMutex );
+    }
+    else
+    {
+        LogError( ( "Failed to acquire mqtt mutex for executing MQTT_Publish"
+                    ",errno=%s",
+                    strerror( errno ) ) );
+    }
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send PUBLISH packet to broker with error = %u.", mqttStatus ) );
+
+        otaRet = OtaMqttPublishFailed;
+    }
+    else
+    {
+        LogInfo( ( "Sent PUBLISH packet to broker %.*s to broker.\n\n",
+                   topicLen,
+                   pacTopic ) );
+    }
+
+    return otaRet;
+}
+
+/*-----------------------------------------------------------*/
+
+static OtaMqttStatus_t mqttUnsubscribe( const char * pTopicFilter,
+                                        uint16_t topicFilterLength,
+                                        uint8_t qos )
+{
+    OtaMqttStatus_t otaRet = OtaMqttSuccess;
+    MQTTStatus_t mqttStatus = MQTTBadParameter;
+
+    MQTTSubscribeInfo_t pSubscriptionList[ 1 ];
+    MQTTContext_t * pMqttContext = &mqttContext;
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    /* Set the QoS , topic and topic length. */
+    pSubscriptionList[ 0 ].qos = qos;
```

Comment:
I understand there may be reasons for not updating the function signature right now, but this assignment is pointless
```suggestion
    ( void ) qos;
```

Reply:
Thanks, addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/11afb1554eb4fcb7cbaf537332c53d27b1085317